document/core/exec/modules.rst

```diff
@@ -277,6 +316,34 @@ New instances of :ref:`functions <syntax-funcinst>`, :ref:`tables <syntax-tablei
    \end{array}
 
 
+.. index:: exception, exception instance, exception address, exception type, function type
+.. _alloc-exn:
+
+:ref:`Exceptions <syntax-exninst>`
+..................................
+
+1. Let :math:`\exn` be the :ref:`exception <syntax-exn>` to allocate for the module :math:`\module`.
+
+2. Let :math:`a` be the first free :ref:`exception address <syntax-exnaddr>` in :math:`S`.
+
+3. Let :math:`\functype` be the :ref:`function type <syntax-functype>` :math:`\module.\MTYPES[\exn.\ETYPE]`.
+
+4. Let :math:`\exninst` be the :ref:`exception instance <syntax-exninst>` :math:`\{ \EITYPE~\functype \}`.
+
+5. Append :math:`\exninst` to the |SEXNS| of :math:`S`.
+
+5. Return :math:`a`.
+
+.. math::
+   \begin{array}{rlll}
+   \allocexn(S, \exn, \module) &=& S', \exnaddr \\[1ex]
```

Comment:
This signature doesn't match its use in embedding.rst, which passes merely an exntype -- which is necessary, because that's outside any module. So I think this function needs to use that, and allocmodule needs adjusting.